backend/app/adapter/graphql/resolver/authquery.go

```diff
@@ -44,6 +44,21 @@ func (v AuthQuery) ChangeLog() (ChangeLog, error) {
 	return newChangeLog(changeLog, lastViewedAt), err
 }
 
+// ListURLs retrieves urls for given user from persistent storage
+func (v AuthQuery) ListURLs() ([]URL, error) {
+	urls, err := v.urlRetriever.GetURLsByUser(*v.user)
```

Comment:
There is a possibility to dereference `nil` pointer.

Reply:
I think if user was nil then it would throw an error in Query resolver

Reply:
@kapillamba4 We have to `nil` check before reaching this line.

Reply:
what error message should be sent if user is nil?

Reply:
"Authentication failed" ?

Reply:
Permission denied.

Reply:
@byliuyang updated